migration/identity-migration/src/main/java/io/camunda/migration/identity/AuthorizationMigrationHandler.java

```diff
@@ -7,133 +7,155 @@
-
-    authorizationMap.forEach(
-        (owner, value) -> {
-          value.forEach(
-              (resourceType, permissionAndResources) -> {
-                final Map<PermissionType, Set<String>> permissions =
-                    permissionAndResources.entrySet().stream()
-                        .flatMap(
-                            e ->
-                                convertPermission(e).stream()
-                                    .map(entry -> new SimpleEntry<>(entry, e.getValue())))
-                        .collect(
-                            Collectors.groupingBy(
-                                SimpleEntry::getKey,
-                                Collectors.flatMapping(
-                                    e -> e.getValue().stream(), Collectors.toSet())));
-                final long ownerKey = getOwnerKeyForUsername(owner);
-                // TODO: this part needs to be revisited
-                //                authorizationService
-                //                    .patchAuthorization(
-                //                        new PatchAuthorizationRequest(
-                //                            ownerKey,
-                //                            PermissionAction.ADD,
-                //                            convertResourceType(resourceType),
-                //                            permissions))
-                //                    .join();
+  protected void process(final List<Authorization> authorizations) {
+    createAuthorizations();
+  }
 
-                final Collection<UserResourceAuthorization> migrated =
-                    permissionAndResources.entrySet().stream()
-                        .flatMap(
-                            entry ->
-                                entry.getValue().stream()
-                                    .map(
-                                        resourceId ->
-                                            new UserResourceAuthorization(
-                                                owner, resourceId, resourceType, entry.getKey())))
-                        .toList();
-                managementIdentityClient.markAuthorizationsAsMigrated(migrated);
-              });
+  private void createAuthorizations() {
+    final var usersEmailByUsername = getUsersEmailByID();
+    final var authorizations = managementIdentityClient.fetchAuthorizations();
+    authorizations.forEach(
+        authorization -> {
+          final var request =
+              new CreateAuthorizationRequest(
+                  getOwnerId(authorization.entityId(), usersEmailByUsername),
+                  convertOwnerType(authorization.entityType()),
+                  authorization.resourceKey(),
+                  convertResourceType(authorization.resourceType()),
+                  convertPermissions(authorization.permissions(), authorization.resourceType()));
+          try {
+            authorizationService.createAuthorization(request).join();
```

Comment:
🔧 as resourceType and ownerType can be UNSPECIFIED, should we rather log (warn) and skip those entities? Same for empty permissions list?

Reply:
got it, effectively that should be unlikely anyway, at most it might be new types added later we don't support.